Telerik.JustMock/Core/Behaviors/RaiseEventBehavior.cs

```diff
@@ -60,18 +61,39 @@ public void Process(object[] invocationArgs, Type declaringType)
 		public static void RaiseEventImpl(object instance, EventInfo evt, object[] args)
 		{
 			if (evt == null)
+			{
 				throw new MockException("Unable to deduce which event was specified in the parameter.");
+			}
 
 			if (args == null)
 			{
 				args = new object[] { null };
 			}
 
-			if (args.Length == 1
-				&& (evt.EventHandlerType.IsGenericType && evt.EventHandlerType.GetGenericTypeDefinition() == typeof(EventHandler<>)
-					|| evt.EventHandlerType == typeof(EventHandler)
-					|| args[0] is EventArgs)
-				)
+			bool shouldInsertEventSender = false;
+
+			if (evt.EventHandlerType.IsGenericType
```

Comment:
I can somewhat understand what's happening here, but I still think it would be beneficial to add a comment stating what is going on and why we're adding the sender instance as the first argument.

Reply:
@mbektchiev, fixed with [df497f3](https://github.com/telerik/JustMockLite/pull/211/commits/df497f3840ec3d36ee347703bd0bb752e44d1f58)